unresolved function is a placeholder node, but Eval was called

I'm not sure what to do with this panic:

```
panic: unresolved function is a placeholder node, but Eval was called [recovered]
        panic: unresolved function is a placeholder node, but Eval was called
```

Here's the stacktrace not including our code:

```
github.com/liquidata-inc/go-mysql-server/sql/expression.(*UnresolvedFunction).Eval(0xc00020ad80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/expression/unresolved.go:128 +0x39
github.com/liquidata-inc/go-mysql-server/sql/parse.columnDefinitionToColumn(0x0, 0xc00030cc00, 0xc000098780, 0x4, 0x4, 0xc0000ccbc0, 0x4, 0x8)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/parse/parse.go:820 +0x374
github.com/liquidata-inc/go-mysql-server/sql/parse.TableSpecToSchema(0x0, 0xc0002009c0, 0xc0002f6000, 0x11f2280, 0xc0000be790, 0x0, 0xfd3660)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/parse/parse.go:769 +0xba
github.com/liquidata-inc/go-mysql-server/sql/parse.convertCreateTable(0xc0001ea1c0, 0xc0002769a0, 0xff90bd, 0x6, 0x351, 0x380)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/parse/parse.go:624 +0x3f
github.com/liquidata-inc/go-mysql-server/sql/parse.convertDDL(0xc0001ea1c0, 0xc000558700, 0x351, 0xc0002769a0, 0x0, 0x0, 0xc000558a80, 0xc00043cbd8)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/parse/parse.go:463 +0x26f
github.com/liquidata-inc/go-mysql-server/sql/parse.convert(0xc0001ea1c0, 0x11fe620, 0xc000276840, 0xc000558700, 0x351, 0x0, 0xc000558a80, 0x351, 0x0)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/parse/parse.go:153 +0x251
github.com/liquidata-inc/go-mysql-server/sql/parse.Parse(0xc0001ea1c0, 0xc000558380, 0x353, 0x0, 0x0, 0x0, 0x0)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/sql/parse/parse.go:125 +0x54a
github.com/liquidata-inc/go-mysql-server.(*Engine).Query(0xc0000982e0, 0xc0001ea000, 0xc000558380, 0x353, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/frew/go/pkg/mod/github.com/liquidata-inc/go-mysql-server@v0.5.1-0.20200508170854-b74c3414aebf/engine.go:116 +0x108
```

This was trying to run the following DDL:

```
CREATE TABLE `job_dismissal` (
  `job_dismissal_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `contact_id` int(10) unsigned NOT NULL,
  `quiz_id` int(10) unsigned DEFAULT NULL,
  `external_job_id` varchar(100) DEFAULT NULL,
  `job_fingerprint` varchar(32) NOT NULL,
  `create_time` timestamp(6) NOT NULL DEFAULT NOW(),
  PRIMARY KEY (`job_dismissal_id`),
  KEY `idx_qzdsmsl_contact_id_dismissal_id` (`contact_id`,`job_dismissal_id`,`job_fingerprint`),
  KEY `idx_qzdsmsl_quiz_id_contact_id` (`quiz_id`,`contact_id`),
  KEY `idx_qzdsmsl_create_time` (`create_time`),
  CONSTRAINT `FK_QZDSMSL_contact_id` FOREIGN KEY (`contact_id`) REFERENCES `contact` (`contact_id`),
  CONSTRAINT `FK_QZDSMSL_quiz_id` FOREIGN KEY (`quiz_id`) REFERENCES `quiz` (`quiz_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT='This table will have INSERTs and DELETEs only';
```

Note that I replaced `CURRENT_TIMESTAMP(6)` with `NOW()` because `CURRENT_TIMESTAMP(6)` errors with `unsupported syntax: current_timestamp(6)`.  I suspect that's much easier to resolve than this.

--- a/enginetest/enginetests.go
+++ b/enginetest/enginetests.go
@@ -657,6 +657,12 @@ func TestCreateTable(t *testing.T, harness Harness) {
 		"c bool, primary key (a))")
 	require.Error(err)
 	require.True(sql.ErrTableAlreadyExists.Is(err))
+
+	_, _, err = e.Query(NewContext(harness), "CREATE TABLE t10(a INTEGER,"+
+			"`create_time` timestamp(6) NOT NULL DEFAULT NOW(),"+
+			"primary key (a))")
+	require.Error(err)
+	require.True(sql.ErrUnsupportedDefault.Is(err))
 }
 
 func TestDropTable(t *testing.T, harness Harness) {

--- a/sql/errors.go
+++ b/sql/errors.go
@@ -65,5 +65,8 @@ var (
 		" this projection with that name. Aliases cannot be used in the same projection they're defined in")
 
 	// ErrInvalidAsOfExpression is returned when an expression for AS OF cannot be used
-	ErrInvalidAsOfExpression = errors.NewKind("Expression %s cannot be used in AS OF")
+	ErrInvalidAsOfExpression = errors.NewKind("expression %s cannot be used in AS OF")
+
+	// ErrUnsupportedDefault return when an expression for a column default is not supported
+	ErrUnsupportedDefault = errors.NewKind("column default %s is not supported")
 )

--- a/sql/parse/parse.go
+++ b/sql/parse/parse.go
@@ -978,6 +978,9 @@ func columnDefinitionToColumn(ctx *sql.Context, cd *sqlparser.ColumnDefinition, 
 		// TODO: this isn't quite right -- some default expressions here (like function calls) need to be stored by the
 		//  implementor and deferred until row insertion time. We can't do that, but we can at least do a better job
 		//  detecting when this happens and erroring out.
+		if !dflt.Resolved() {
+			return nil, sql.ErrUnsupportedDefault.New(dflt.String())
+		}
 		defaultVal, err = dflt.Eval(ctx, nil)
 		if err != nil {
 			return nil, ErrUnsupportedFeature.New("column defaults must be evaluable at schema modification time")

